### PR TITLE
Enable PDK (validation, testing)

### PR DESCRIPTION
**Pull request by bittner**
As mentioned in #9, a few changes can make it possible to use Puppet's new [PDK](https://github.com/puppetlabs/pdk) with this module.

I've also fixed the linter complaints `pdk validate` has reported.

### CODE REVIEW

**Comment by bittner** [comment]
The Travis configuration now uses the PDK commands and the beautiful Travis build stages feature for enhanced readability.

We could specify a list of Rubys or RVMs in addition for running a build matrix, e.g.

```yaml
rvm:
  - 2.3
  - 2.4
```

More details in the [related docs](https://docs.travis-ci.com/user/languages/ruby/).

You probably know better what would be sensible to test against.

**Comment by bittner** [comment]
Note that PR #16 contains changes that will conflict with this PR.

Specifically, the [RSpec tests for Skype](https://github.com/edestecd/puppet-software/pull/16/files#diff-05b6c41b1ec163a989fa0ad4781b673aR18) require the old hash rocket syntax and no trailing comma after the last item in the `with` clause. This should be changed when merging to:

```ruby
          it {
            is_expected.to contain_apt__source('skype-stable')
              .with(
                location: 'https://repo.skype.com/deb',
                release: 'stable',
                repos: 'main',
                architecture: 'amd64',
              )
          }
```

**Comment by bittner** [comment]
This PR is now ready for merging.

**Comment by bittner** [comment]
@edestecd Any chance of getting this merged?

**Comment by bittner** [comment]
:bell: **ping!**

Please merge!

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
   include:
-  - rvm: 2.1.9
-    env: PUPPET_GEM_VERSION="~> 4.0" STRICT_VARIABLES="yes"
-notifications:
-  email: false
+    - stage: validate
+      script: pdk validate 2> /dev/null
+
+    - stage: test
+      script: pdk test unit 2> /dev/null
```
Comment (bittner) [inline .travis.yml]: The redirection of the progress output may not be needed anymore (or be replaced by a [command line options](https://puppet.com/docs/pdk/1.x/pdk_reference.html#reference-432)) for the latest versions of the PDK. But that can also be done once this PR is merged.

**Comment by bittner** [comment]
@edestecd Any chance to get this merged?

**Comment by edestecd** [comment]
@bittner Sorry, thanks for bugging me about it. This module needs some serious love. I finally got around to learning PDK and converted some of our modules already. Can you use a newer version of PDK and use `pdk convert`: https://puppet.com/docs/pdk/1.x/pdk_reference.html#pdk-convert-command

PDK version 1.4.1 or 1.5.0 will do. I think my other modules are pinned at 1.4.1, so thats probably best: https://github.com/edestecd/puppet-mariadb/blob/master/metadata.json#L74


Also can you use this .sync.yml? https://github.com/edestecd/puppet-mariadb/blob/master/.sync.yml
Put it in prior to running pdk convert to get our preferred template settings.
I'll update the secure password for puppetforge after its merged.

**Comment by bittner** [comment]
I don't have access to the `comsolit` repository anymore (stopped working with that client). It would be easier for me if you could merge the PR "as is", and we deal with refactoring things afterwards. I believe the build should still pass.

**Comment by edestecd** [comment]
If you want to do the PDK convert also, let me know and i'll wait

**Comment by bittner** [comment]
> If you want to do the PDK convert also, let me know and i'll wait

No, go on with the merging. Thank you!

**Comment by bittner** [comment]
Thank you! :+1:

**Comment by bittner** [comment]
The [build on `master` fails now](https://travis-ci.org/edestecd/puppet-software/jobs/377396606) related to Windows tooling, if I'm not totally mistaken:

```
Could not find class ::chocolatey
```
Not sure what's going on. :confused:

**Comment by edestecd** [comment]
I'll look in a bit. I'm going to update pdk in a few.

**Comment by bittner** [comment]
Could it be that [chocolately](https://github.com/chocolatey/choco) needs to be added to the [.fixtures.yml](https://github.com/edestecd/puppet-software/blob/master/.fixtures.yml)?

**Comment by edestecd** [comment]
more than likely

**Comment by edestecd** [comment]
Whew, there were a lot of windows issues I had to fix. I suspect the tests were not actually running against windows before. Probably the facts gem did not contain facts for the versions of windows specified in out metadata.json. Anyways its all fixed up now.

**Pull request merged**